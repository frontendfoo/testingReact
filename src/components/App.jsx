import React, { memo } from 'react'

const App = ({ handleSetName, name, disabled, handleSubmitName }) => (
  <div>
    <h1>login</h1>
    <input
      type='text'
      className='w-full mt-8'
      placeholder='Enter your name here...'
      onChange={handleSetName}
      value={name}
    />
    <div>
      <button className='mt-12' disabled={disabled} onClick={handleSubmitName}>
        Login!
      </button>
    </div>
  </div>
)
export default memo(App)
