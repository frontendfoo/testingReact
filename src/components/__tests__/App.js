// React
import React from 'react'

// Test Renderer
import ShallowRenderer from 'react-test-renderer/shallow'

// Component
import App from '../App'

const renderer = new ShallowRenderer()

describe('The App Component', () => {
  const propVariations = [
    {},
    {
      handleSetName: '',
      name: '',
      disabled: false,
      handleSubmitName: ''
    },
    {
      handleSetName: 'Function setName',
      name: 'Test Name',
      disabled: true,
      handleSubmitName: 'Function submitName'
    }
  ]

  propVariations.forEach((props, i) => {
    const result = renderer.render(<App {...props} />)
    it(`match snapshot with prop variation ${++i}`, () => {
      expect(result).toMatchSnapshot()
    })
  })
})
