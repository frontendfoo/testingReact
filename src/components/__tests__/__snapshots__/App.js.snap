// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Component match snapshot with prop variation 1 1`] = `
<div>
  <h1>
    login
  </h1>
  <input
    className="w-full mt-8"
    placeholder="Enter your name here..."
    type="text"
  />
  <div>
    <button
      className="mt-12"
    >
      Login!
    </button>
  </div>
</div>
`;

exports[`The App Component match snapshot with prop variation 2 1`] = `
<div>
  <h1>
    login
  </h1>
  <input
    className="w-full mt-8"
    onChange=""
    placeholder="Enter your name here..."
    type="text"
    value=""
  />
  <div>
    <button
      className="mt-12"
      disabled={false}
      onClick=""
    >
      Login!
    </button>
  </div>
</div>
`;

exports[`The App Component match snapshot with prop variation 3 1`] = `
<div>
  <h1>
    login
  </h1>
  <input
    className="w-full mt-8"
    onChange="Function setName"
    placeholder="Enter your name here..."
    type="text"
    value="Test Name"
  />
  <div>
    <button
      className="mt-12"
      disabled={true}
      onClick="Function submitName"
    >
      Login!
    </button>
  </div>
</div>
`;
